### ISSUE TITLE
[BUG] Abstract and interface class trace_method hooks no longer tracing

### ISSUE THREAD

**Issue by victorjkhalaf** [DataDog/dd-trace-php#1921]
### Bug description

A change in version 0.76 seems to have changed how the trace_method hooking works. I have discussed this in a case with Support but I am posting it here for public awareness for anyone experiencing a change in how their manual tracing works.

This may be more of a concern for long running process tracing where common interface and shared abstract class based jobs handling is often used. 
We had manual trace_method() calls that targeted abstract classes' methods, allowing us to reliably trace any job or plugin (class based) we were running through the long running process. They stopped tracing after v0.76 and it seems the DD team did some change to the hooking where "if the parent method is interface or abstract it is not traced" (quote from support).

For example this will not work anymore

```php
\DDTrace\trace_method(
        AbstractJob::class,
        'process',
        function (\DDTrace\SpanData $span, array $args, $retval, $exception) {
        }
);
```
The alternative for now is to manual trace each class, which could be quite a lot of them and a lot of code blocks to copy paste.

This may be apparently something they will try to restore, but until then I think it's valuable to mention it to all.

I understand this may not be preferably labeled as BUG, feel free to change that label. From our perspective functionality stopped working, so it felt like a bug.

### PHP version

8.2

### Tracer version

0.83.0

### OS info

Alpine 3.16

**Comment by bwoebi** [comment]
Hey @victorjkhalaf,

In fact, I already have a PR for that: #1914 - we're looking at including it in the next release :-)

It was _not_ intended to have this stop working, but I was not aware of this working in the first place when I broke it (as we had no tests for that). Sorry for this, but soon it will be fixed!

**Comment by victorjkhalaf** [comment]
That's wonderful to hear and I am glad that my case helped get this identified and promptly resolved. :) I'll be keeping a watch on the releases.

**Comment by bwoebi** [comment]
The release has been released now - 0.85.0 has support for this.

**Comment by Gayathri2102** [comment]
Is there an example on how to use this in a kotlin project? I'm still now able to trace the abstract methods.
